### PR TITLE
If repo exist, try to pull before recloning it.

### PR DESCRIPTION
**Pull request by rlaverde**
Fixes: #846

This is missing a test, I'm not sure how this could be tested, someone could give a hint?

### CODE REVIEW

**Comment by hackebrot** [comment]
Hi @rlaverde! 👋 

Thank you for your contribution! I wonder if we should prompt the user whether they want to perform a git pull first. Otherwise we overwrite their local template and they won't even know.

What do you think, @audreyr @pydanny @michaeljoseph?

**Comment by rlaverde** [comment]
> I wonder if we should prompt the user whether they want to perform a git pull first. Otherwise we overwrite their local template and they won't even know.

I'm not sure, I didn't add a prompt because I think that normally people will want to use the latest version of the cookie(template)

**Comment by hackebrot** [comment]
I'm not sure, right now as a user you can decide not to modify the local template when using the CLI. With this change we take that option away from them.

🔔 @michaeljoseph

**Comment by michaeljoseph** [comment]
Yeah, I think I'm with @hackebrot on this one.  It would make more sense to me to add this functionality when the user declines deletion and re-cloning, since we currently just exit (which has always seemed surprising).
I think we should ask the user if they'd like to `pull`, and then use the existing repo if they decline.

File: cookiecutter/vcs.py [hunk 1]
```diff
@@ -39,6 +39,7 @@ def prompt_and_delete_repo(repo_dir, no_input=False):
     else:
         question = (
             "You've cloned {} before. "
+            "And It wasn't possible to pull new changes. "
```
Comment (theodesp) [inline cookiecutter/vcs.py]: Lower case `It` here --> `it'

File: tests/vcs/test_clone.py [hunk 2]
```diff
@@ -73,10 +73,6 @@ def test_clone_should_abort_if_user_does_not_want_to_reclone(mocker, tmpdir):
         side_effect=SystemExit,
         autospec=True
     )
```
Comment (theodesp) [inline tests/vcs/test_clone.py]: I would like more tests here covering those scenarios you mentioned

**Comment by theodesp** [comment]
@rlaverde You need also to fix the CI errors 👍

**Pull request closed without merging**